pages/human-donors/[uuid].js

```diff
@@ -125,6 +138,27 @@ export async function getServerSideProps({ params, req }) {
           filterErrors: true,
         })
       : [];
+    const phenotypicFeatures = donor.phenotypic_features
+      ? await request.getMultipleObjects(donor.phenotypic_features, null, {
+          filterErrors: true,
+        })
+      : [];
+    const phenotypeTermsList = phenotypicFeatures
+      ? await request.getMultipleObjects(
+          phenotypicFeatures.map((phenotype) => phenotype.feature)
+        )
+      : [];
+    const phenotypeTerms = phenotypeTermsList.reduce((obj, el) => {
+      obj[el["@id"]] = el;
+      return obj;
+    }, {});
+
+    const materializedPhenotypicFeatures = phenotypicFeatures.map((feature) => {
+      const term = phenotypeTerms[feature.feature];
+      feature.feature = term;
+      return feature;
+    });
```

Comment:
I’m allergic to `.map()` loops with side effects. I think, after retrieving all the `phenotypic_features` objects, and all their `feature` objects (and I would dedupe the `feature` objects list before doing the `getMultipleFeatures()` request in case many `phenotypic_feature` objects refer to the same `PhenotypeTerm` object — no idea if that’ll be the case, but just in case). Then just send both these arrays to props without tying them together in any way.

Then the `HumanDonor` component takes both arrays. Pass both arrays to `<PhenotypicFeatureTable>`, and inside that pass the `features` array in the `meta` property to `<SortableGrid>`. Then in the `phenotypicFeaturesColumns[x].display` for the `feature` column, get the meta in the second parameter to that function (see the `<SortableGrid>` documentation Markdown file) and then that `display(source, meta)` function searches the `features` property for the item being displayed. It’ll be OK to do a search of the `features` array for each row — it’s highly unlikely to be a performance bottleneck.